tests/test_endorsement.py

```diff
@@ -0,0 +1,104 @@
+         session.add(endorser)
+         endorsee = TapirUsers(first_name='Aspen', last_name='Early', policy_class=2, email='ase12@cornell.edu')
+         session.add(endorsee)
+         endorsee_sus = TapirUsers(first_name='Bobby', last_name='NoPapers', policy_class=2, email='bob@scamy.example.com')
+         endorsee_sus.demographics = Demographics(flag_suspect=1, archive='cs', subject_class='CR')
+         session.add(endorsee_sus)
+
+         req1 = EndorsementRequests(endorsee=endorsee,
+                                    secret='end_sec_0',
+                                    archive='cs', subject_class='CR',
+                                    issued_when=datetime.now(),
+                                    flag_valid=1, point_value=10)
+         req1.endorsement = Endorsements(endorser=endorser, endorsee=endorsee,                                     archive='cs', subject_class='CR',)
+
+         req2 = EndorsementRequests(endorsee=endorsee_sus,
+                                    secret='end_sec_1',
+                                    archive='cs', subject_class='CR',
+                                    issued_when=datetime.now(),
+                                    flag_valid=1, point_value=10)
+         req2.endorsement = Endorsements(endorser=endorser,                   endorsee=endorsee_sus    ,              archive='cs', subject_class='CR',)
+
+         req3 = EndorsementRequests(endorsee=endorsee,
+                                    secret='end_sec_2',
+                                    archive='cs', subject_class='CV',
+                                    issued_when=datetime.now(),
+                                    flag_valid=1, point_value=-10)
+         req3.endorsement = Endorsements(endorser=endorser, endorsee=endorsee         ,                            archive='cs', subject_class='CV')
+
+         req4 = EndorsementRequests(endorsee=endorsee,
+                                    secret='end_sec_3',
+                                    archive='cs', subject_class='DL',
+                                    issued_when=datetime.now() - timedelta(days=4),
+                                    flag_valid=1, point_value=10)
+         req4.endorsement = Endorsements(endorser=endorser,                  endorsee=endorsee  ,                 archive='cs', subject_class='DL')
+
+         req5 = EndorsementRequests(endorsee=endorsee,
+                                    secret='end_sec_5',
+                                    archive='cs', subject_class='ET',
+                                    issued_when=datetime.now() - timedelta(days=50),
+                                    flag_valid=1, point_value=10)
+         req5.endorsement = Endorsements(endorser=endorser,                  endorsee=endorsee  ,                 archive='cs', subject_class='ET')
+
+         session.commit()
+         def as_html_parts(id):
+             return f">{id}</a>";
```

Comment:
The same code is repeated in the factory.py so that there is data when you run the development server. Any way to eliminate this duplication?